File: creacion/builder/python/solution/src/main/classes/persona.py
#!/usr/bin/python3
# coding: utf-8

class Persona:

    def __init__(self):

        self._nombre = None
        self._edad = None
        self._municipio = None
        self._colegio = None
        self._lugarTrabajo = None

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, nombre):
        self._nombre = nombre

    @property
    def edad(self):
        return self._edad

    @edad.setter
    def edad(self, edad):
        self._edad = edad


    @property
    def municipio(self):
        return self._municipio


    @municipio.setter
    def municipio(self, municipio):
        self._municipio = municipio


    @property
    def colegio(self):
        return self._colegio


    @colegio.setter
    def colegio(self, colegio):
        self._colegio = colegio


    @property
    def lugarTrabajo (self):
        return self._lugarTrabajo


    @nombre.setter
    def lugarTrabajo (self, lugarTrabajo ):
        self._lugarTrabajo  = lugarTrabajo

File: creacion/builder/java/patterns_2_builder_ejercicio/src/main/java/org/formacion/builder/Persona.java
package org.formacion.builder;

public class Persona {

	private String nombre;
	private int edad;
	private String municipio;
	private String colegio;
	private String lugarTrabajo;
	
	private Persona() {}
	
	
	public static class Builder {
		private Persona persona;
		
		public Builder(String nombre) {
			persona = new Persona();
			persona.nombre = nombre;
		}
		
		public Builder setMunicipio (String municipio) {
			persona.municipio = municipio;
			return this;
		}
		
		public Builder setMayor(int edad, String lugarTrabajo) {
			if (edad < 18) throw new IllegalArgumentException("es menor de edad " + edad);
			persona.edad = edad;
			persona.lugarTrabajo = lugarTrabajo;
			persona.colegio = null;
			return this;
		}
		
		public Builder setMenor(int edad, String colegio) {
			if (edad >= 18) throw new IllegalArgumentException("es mayor de edad " + edad);
			persona.edad = edad;
			persona.colegio = colegio;
			persona.lugarTrabajo = null;
			return this;
		}
	
		public Persona build() {
			return persona;
		}

	}
	
}
